githubdelivery/src/main/java/io/dockstore/githubdelivery/GithubDeliveryHelper.java

```diff
@@ -0,0 +1,62 @@
+package io.dockstore.githubdelivery;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.JsonSyntaxException;
+import io.dockstore.openapi.client.model.InstallationRepositoriesPayload;
+import io.dockstore.openapi.client.model.PushPayload;
+import io.dockstore.openapi.client.model.ReleasePayload;
+import static io.dockstore.utils.ExceptionHandler.exceptionMessage;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+
+/**
+ * Helper for deserializing GitHub events in JSON to Java objects, as well as some logging
+ * methods.
+ */
+final class GithubDeliveryHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GithubDeliveryHelper.class);
+    
+    private static final ObjectMapper MAPPER = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    private GithubDeliveryHelper() {
+    }
+
+    static PushPayload getGitHubPushPayloadByKey(String eventType, String body, String key) throws IOException, NoSuchKeyException {
+        return getPayloadByKey(eventType, body, key, PushPayload.class);
+    }
+
+    static InstallationRepositoriesPayload getGitHubInstallationRepositoriesPayloadByKey(String eventType, String body, String key) throws IOException, NoSuchKeyException {
+        return getPayloadByKey(eventType, body, key, InstallationRepositoriesPayload.class);
+    }
+
+    static ReleasePayload getGitHubReleasePayloadByKey(String eventType, String body, String key) throws IOException, NoSuchKeyException {
+        return getPayloadByKey(eventType, body, key, ReleasePayload.class);
+    }
+
+    static <T> T getPayloadByKey(String eventType, String body, String key, Class<T> clazz) throws IOException,
+        NoSuchKeyException {
+        try {
+            T payload = MAPPER.readValue(body, clazz);
+            if (payload == null) {
+                logReadError(eventType, key);
+            }
+            return payload;
+        } catch (JsonSyntaxException e) {
+            exceptionReadError(e, eventType, key);
+        }
+        return null;
+    }
+
+    static void logReadError(String eventType, String key) {
```

Comment:
Are these two essentially private? Are  they used anywhere else or more than once for that matter?

Reply:
They are used in GitHubDeliveryS3Client